src/promptflow/promptflow/batch/_batch_inputs_processor.py

```diff
@@ -260,13 +269,20 @@ def apply_inputs_mapping(
     """
     result = {}
     notfound_mapping_relations = []
+    # Add property on operation context, which is used to differentiate the input source of a run.
+    # Set input source to user provided by default.
+    operation_context = OperationContext.get_instance()
+    operation_context.input_source = InputSource.UserProvided.name
     for map_to_key, map_value in inputs_mapping.items():
         # Ignore reserved key configuration from input mapping.
         if map_to_key == LINE_NUMBER_KEY:
             continue
         if not isinstance(map_value, str):  # All non-string values are literal values.
             result[map_to_key] = map_value
             continue
+        # If map_value is a previous run output, set input source to previous run.
+        if map_value.startswith("${run.outputs"):
```

Comment:
Checking the prefix of `map_value` seems to be a little hacky and not straightforward. Is there a better way?